gnosis/util/util.py

```diff
@@ -1,5 +1,18 @@
 from typing import Any, Iterable, List
 
+try:
+    from functools import cached_property  # noqa F401
+except ImportError:
```

Comment:
Out of curiosity... why do we have this fallback, is it to support old Python versions?

Reply:
To support old python versions. It was introduced in pyhon 3.8: https://docs.python.org/3/library/functools.html#functools.cached_property